extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/VertxTracingAdapter.java

```diff
@@ -0,0 +1,289 @@
+package io.quarkus.opentelemetry.runtime.tracing.vertx;
+
+import static io.opentelemetry.context.Context.current;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_CLIENT_IP;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_FLAVOR;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_HOST;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_METHOD;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_SCHEME;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_STATUS_CODE;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_TARGET;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_URL;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.HTTP_USER_AGENT;
+import static io.vertx.core.http.HttpHeaders.CONTENT_LENGTH;
+import static io.vertx.core.http.HttpHeaders.USER_AGENT;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.vertx.core.Context;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.core.impl.ContextInternal;
+import io.vertx.core.spi.VertxTracerFactory;
+import io.vertx.core.spi.tracing.SpanKind;
+import io.vertx.core.spi.tracing.TagExtractor;
+import io.vertx.core.spi.tracing.VertxTracer;
+import io.vertx.core.tracing.TracingOptions;
+import io.vertx.core.tracing.TracingPolicy;
+
+public class VertxTracingAdapter extends TracingOptions implements VertxTracer<Span, Span>, VertxTracerFactory {
+    private static final String SCOPE_KEY = VertxTracingAdapter.class.getName() + ".scope";
+    private static final String SPAN_KEY = VertxTracingAdapter.class.getName() + ".activeSpan";
+    private static TextMapPropagator TEXT_MAP_PROPAGATOR;
+
+    private Tracer tracer;
+
+    public void init() {
+        Instance<Tracer> tracerInstance = CDI.current().select(Tracer.class);
+        if (tracerInstance.isResolvable()) {
```

Comment:
what happens if it isn't?

Reply:
In theory it will never reach that point, as it would mean the `Producer` for `Tracer` is not present but this class is.

Enable/Disable tracing should mean both are present or both are missing.

Reply:
ok. so why bother checking? You don't test for null anywhere else in here. If you're going to bother checking, you need to handle what happens if it isn't resolvable. Does that mean you throw? You turn it off? Cause an NPE is waiting (however unlikely, I will grant)